### PR TITLE
Close #5. Add initial collision detection.

### LINKED ISSUE #5: Jeepney can collide with objects
When the jeepney collides with an object, the object should get knocked off of the screen, and the jeepney should take damage.

### PR DESCRIPTION
**Pull request by allindow**
Close #5 Jeepney can collide with objects

### CODE REVIEW

**Comment by allindow** [comment]
@ryanflach ......you're going to love what happens to the jeepney based on its health 😆 

They're so much that needs to be refactored into separate methods, but it's functional. I have the obstacles flying off the screen diagonally after collision but couldn't figure out how to make it actually look like they're flying off - visually they're just disappearing. If we can slow it down then having them move off into different directions depending on where they're hit shouldn't be too difficult. 

I didn't add any tests as I'm still not sure how to test for this kind of thing, but I ran the suite for existing tests and everything was passing still.

File: lib/jeepney.js [hunk 1]
```diff
@@ -42,4 +42,12 @@ Jeepney.prototype.draw = function(context) {
   context.drawImage(this.img, this.x, this.y, this.width, this.height);
 };
 
+Jeepney.prototype.isColliding = function(obstacle){
+  return (
+    (this.x > (obstacle.x - obstacle.width)) &&
+    ((this.y - this.height) < obstacle.y && obstacle.y < this.y) &&
+    (this.x < obstacle.x)
+  )
```
Comment (ryanflach) [inline lib/jeepney.js]: JSHint would love a semi-colon here.

File: lib/game.js [hunk 2]
```diff
@@ -27,6 +27,9 @@ Game.prototype.update = function() {
   var lastbackgroundObject = this.backgroundObjects[this.backgroundObjects.length - 1];
   var firstObstacle = this.obstacles[0];
   var lastObstacle = this.obstacles[this.obstacles.length - 1];
+  var jeepney = this.jeepney;
+  var obstacles = this.obstacles;
+  var motorcycle = this.motorcycle;
```
Comment (ryanflach) [inline lib/game.js]: I think we can remove lines 31-32. Both of these variables are defined but not used.

File: lib/game.js [hunk 3]
```diff
@@ -46,6 +49,19 @@ Game.prototype.update = function() {
...
     obstacle.update();
+    if (jeepney.isColliding(obstacle)){
+      jeepney.health --;
+      if (jeepney.health <= 2 ){
+        jeepney.img.src = '/assets/images/jeepney-damage3.png'
+      } else if (jeepney.health <= 3){
+        jeepney.img.src = '/assets/images/jeepney-damage2.png'
+      } else {
+        jeepney.img.src = '/assets/images/jeepney-damage1.png'
+      }
```
Comment (ryanflach) [inline lib/game.js]: JSHint is asking for a semi-colon at the end of lines 55, 57, and 59, please.

**Review by ryanflach — CHANGES_REQUESTED** [review]
@allindow This looks great overall! I LOVE what you did with the Jeepney image progression. There are some kinks we'll need to work out with the collision detection (currently is seems to only respond if it's a head-on collision with the jeepney or the motorcycle is directly underneath the jeepney), but we can work on that later down the road (same with obstacles flying off). Great job with this. If you'd like to go ahead and fix the few very minor syntax issues brought up by JSHint I will merge.

**Comment by allindow** [comment]
changes made

**Review by ryanflach — APPROVED** [review]

**Comment by ryanflach** [comment]
@allindow Looks great. Merging.

**Pull request merged**